Pushbullet email (fix)

todschmidt opened the pull request:
## Description:

- Enable email notifications for all push types, previously only worked for notes.
- Enable list as a data parameter, to pass lists as well as files,
- Added super basic unittest, can't do much more without API KEY, emailed pushbullet to see if
  they make a test api available, will add more tests if they do.

**Related issue (if applicable):** fixes #11198

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io# N/A

## Example entry for `configuration.yaml` (if applicable):
```yaml
test_pushbullet_file:
  sequence:
    service: notify.pushbullet
    data:
      title: Test File Title
      message: Test File
      target:
        - "device/User's iPhone"
        - email/user@host.net
      data:
        file: /tmp/tv_check_doc.jpg

test_pushbullet_note:
  sequence:
    service: notify.pushbullet
    data:
      title: Test Note Title
      message: Test Note

test_pushbullet_list:
  sequence:
    service: notify.pushbullet
    data:
      title: Test List Title
      message: Test List
      data:
        list:
          - item1
          - item2

test_pushbullet_template_list:
  sequence:
    service: notify.pushbullet
    data_template:
      title: Test Template List Title
      message: Test Template List
      data:
        list:
          - "{{ testvar| default('item3') }}"
          - "{{ testvar | default('item4') }}"
```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

maxlaverse committed:
Fix webdav calendar schema (#11185)

jannau committed:
homematic: add username and password to interface config schema (#11214)

Fixes #11191, the json-rpc name resolving method requires user account
and password.

OverloadUT committed:
Fix detection of if a negative node is in use (#11255)

* Fix detection of if a negative node is in use

Fix a problem where every negative node gets detected as in-use. Code was not checking the correct property.

* Allow protected access

pvizeli committed:
Bugfix homematic available modus (#11256)

CTLS committed:
Fix inverted sensors on the concord232 binary sensor component (#11261)

* Fix inverted sensors on the concord232 binary sensor component

* Changed from == Tripped to != Normal

ziotibia81 committed:
Fix handling zero values for state_on/state_off (#11264)

veleek committed:
Fix async IO in Sesame lock component. (#11054)

* Call update on Sesame devices to cache initial state

* Switch to using async_add_devices

* Fix line length

* Fix Lint errors

* Fix more Lint errors

* Cache pysesame properties

* Updates from CR feedback

maxlaverse committed:
Fix allday events in custom_calendars (#11272)

OddBloke committed:
Fix webostv select source (#11227)

* Fix reuse of variable name

This should fix #11224.

* Add tests for LgWebOSDevice.select_source

andreacampi committed:
Support multiple Hue bridges with lights of the same id (#11259)

* Improve support for multiple Hue bridges with lights that have the same id.

The old code pre-refactoring kept a per-bridge list of lights in a closure; my refactoring moved that to hass.data, which is convenient but caused them to conflict with each other.

Fixes #11183

* Update test_hue.py

rwa committed:
Fix unpredictable entity names in concord232 binary_sensor (#11292)

OverloadUT committed:
Fix leak sensors always showing Unknown until Wet (#11313)

Leak sensors were using the "wet" node as a negative node, which prevented them from ever gettng a Dry status unless the user pressed the button on the hardware after every Hass reboot.

This change ignores the Wet node, as it is just always the exact inverse of the Dry node. We don't need to watch both.

pvizeli committed:
Don't block on sevice call for alexa (#11358)

* Don't block on sevice call for alexa

* fix tests

todschmidt committed:
Added email to all pushbullet notification types

todschmidt committed:
Added email to all pushbullet notification types

todschmidt committed:
Added very basic unittest

todschmidt committed:
Added very basic unittest

todschmidt committed:
lint fixes

todschmidt committed:
lint fixes

todschmidt committed:
Added support for pushing list

todschmidt committed:
Added support for pushing list

todschmidt committed:
pylint changes

todschmidt committed:
pylint changes

todschmidt committed:
Update pushbullet.py

todschmidt committed:
Update pushbullet.py

todschmidt committed:
Update pushbullet.py

todschmidt committed:
Update pushbullet.py

DanNixon committed:
Reverts unit conversions in TP-Link bulb (#11381)

Reverts energy and power unit conversions added in #10979 as they break
early devices.

A proper fix should be implemented in the pyhs100 library which should
return common units across all devices/firmwares.

todschmidt committed:
indentation

todschmidt committed:
indentation

balloob committed:
Clean up Google Assistant (#11375)

* Clean up Google Assistant

* Fix tests

jeroenterheerdt committed:
Adding support for Egardia / Woonveilig version GATE-03 (#11397)

Kane610 committed:
Add deCONZ component (#10321)

* Base implementation of component, no sensors yet

* Added senor files

* First fully working chain of sensors and binary sensors going from hardware in to hass

* Clean up

* Clean up

* Added light platform

* Turning lights on and off and set brightness now works

* Pydeconz is now a proper pypi package
Stop sessions when Home Assistant is shutting down
Use a simpler websocket client

* Updated pydocstrings
Followed recommendations from pylint and flake8

* Clean up

* Updated requirements_all.txt

* Updated Codeowners to include deconz.py
Also re-added the Axis component since it had gotten removed

* Bump requirement

* Bumped to v2
Reran script/gen_requirements

* Removed global DECONZ since it wasn't relevant any more

* Username and password is only relevant in the context of getting a API key

* Add support for additional sensors

* Added support for groups

* Moved import of component library to inside of methods

* Moved the need for device id to library

* Bump pydeconz to v5

* Add support for colored lights

* Pylint and flake8 import improvements

* DATA_DECONZ TO DECONZ_DATA

* Add support for transition time

* Add support for flash

* Bump to v7

* ZHASwitch devices will now only generate events by default, instead of being a sensor entity

* Clean up

* Add battery sensor when device signals through an event

* Third-party library communicates with service

* Add support for effect colorloop

* Bump to pydeconz v8

* Same domain everywhere

* Clean up

* Updated requirements_all

* Generated API key will now be stored in a config file

* Change battery sensor to register to callback since library now supports multiple callbacks
Move DeconzEvent to hub
Bump to v9

* Improve entity attributes

* Change end of battery name to battery level
No need for static icon variable when using battery level helper

* Bump requirement to v10

* Improve pydocstring for DeconzEvent
Rename TYPE_AS_EVENT to CONF_TYPE_AS_EVENT

* Allow separate brightness to override RGB brightness

* Expose device.reachable in entity available property

* Bump requirement to 11 (it goes up to 11!)

* Pylint comment

* Binary sensors don't have unit of measurement

* Removed service to generate API key in favor of just generating it as a last resort of no API key is specified in configuration.yaml or deconz.conf

* Replace clear text to attribute definitions

* Use more constants

* Bump requirements to v12

* Color temp requires xy color support

* Only ZHASwitch should be an event

* Bump requirements to v13

* Added effect_list property

* Add attribute to battery sensor to easy find event id

* Bump requirements to v14

* Fix hound comment

* Bumped requirements_all information to v14

* Add service to configure devices on deCONZ

* Add initial support for scenes

* Bump requirements to v15

* Fix review comments

* Python doc string improvement

* Improve setup and error handling during setup

* Changed how to evaluate light features

* Remove 'ghost' events by not triggering updates if the signal originates from a config event
Bump requirement to v17

* Fix pylint issue by moving scene ownership in to groups in requirement pydeconz
Bump requirement to v18

* Added configurator option to register to deCONZ when unlocking gateway through settings
Bump requirement to v20

* Improve async configurator

* No user interaction for deconz.conf

* No file management in event loop

* Improve readability of load platform

* Fewer entity attributes

* Use values() instead of items() for dicts where applicable

* Do one add devices per platform

* Clean up of unused attributes

* Make sure that discovery info is not None

* Only register configure service and shutdown service when deconz has been setup properly

* Move description

* Fix lines longer than 80

* Moved deconz services to a separate file and moved hub to deconz/__init__.py

* Remove option to configure switch as entity

* Moved DeconzEvent to sensor since it is only Switch buttonpress that will be sent as event

* Added support for automatic discovery of deconz
Thanks to Kroimon for adding support to netdisco

* Use markup for configuration description

* Fix coveragerc

* Remove deCONZ support from Hue component

* Improved docstrings and readability

* Remove unnecessary extra name for storing in hass.data, using domain instead

* Improve readability by renaming all async methods
Bump to v21 - improved async naming on methods

* Fix first line not being in imperative mood

* Added logo to configurator
Let deconz.conf be visible since it will be the main config for the component after initial setup

* Removed bridge_type from new unit tests as part of removing deconz support from hue component

* Capitalize first letters of Battery Level

* Properly update state of sensor as well as reachable and battery
Bump dependency to v22

* Fix flake8 Multi-line docstring closing quotes should be on a separate line

* Fix martinhjelmares comments

Bump dependency to v23
Use only HASS aiohttp session
Change when to use 'deconz' or domain or deconz data
Clean up unused logger defines
Remove unnecessary return values
Fix faulty references to component documentation
Move callback registration to after entity has been initialized by HASS
Less inception style on pydocs ;)
Simplify loading platforms by using a for loop
Added voluptous schema for service
Yaml file is for deconz only, no need to have the domain present
Remove domain constraint when creating event title

balloob committed:
Fix broken tests (#11395)

* Do not leave remember the milk config file behind

* Fix exception in service causing service timeout

* Change max service timeout to 9 to catch services timing out

* Fix Google Sync service test

* Update and pin test requirements

OttoWinter committed:
Upgrade pychromecast to 1.0.3 (#11410)

pavoni committed:
Bump pywemo to fix request include problems. (#11401)

balloob committed:
Log exceptions that happen during service call (#11394)

* Log exceptions that happen during service call

* Lint

balloob committed:
Update frontend to 20180102.0

notoriousbdg committed:
Move IMAP Email Content body to an attribute (#11096)

* Move IMAP Email Content body to an attribute

* Fix variable names

masarliev committed:
Notify webos timeout error fix (#11027)

* use smaller icon

* add timeout option

* remove default icon, remove timeout option

* add image again

ciotlosm committed:
Removed status block to allow https://github.com/home-assistant/home-assistant-polymer/pull/766 with no impact (#11345)

DanNixon committed:
More support for availability reporting on MQTT components (#11336)

* Abstract MQTT availability from individual components

- Moved availability topic and payloads to MQTT base schema.
- Updated components that already report availability:
  - Switch
  - Binary sensor
  - Cover

* Add availability reporting to additional MQTT components

- Light
- JSON light
- Template light
- Lock
- Fan
- HVAC
- Sensor
- Vacuum
- Alarm control panel

* Annotate MQTT platform coroutines

tinloaf committed:
Do not purge the most recent state for an entity (#11039)

* Protect states that are the most recent states of their entity

* Also protect events

* Some documentation

* Fix SQL

amelchio committed:
Performance optimization of TP-Link switch (#11416)

tomaszduda23 committed:
Fixing OpenWeatherMap Sensor. Current weather is 'unknown' if forecast: false. It was reported as #8640. (#11417)

amelchio committed:
Silence redundant warnings about slow setup (#11352)

* Downgrade slow domain setup warning

* Revert "Downgrade slow domain setup warning"

This reverts commit 64472c006bb553c6bb75a024384361adad50d565.

* Remove warning for entity components

* Remove lint

* Fix original test with extra call

akatrevorjay committed:
Account for User-Agent being non-existent, causing a TypeError (#11064)

* Account for User-Agent being non-existent, causing a TypeError

* Actually fix case of no user-agent with last resort

* Return es5 as last resort

* Update __init__.py

* Update __init__.py

Danielhiversen commented on homeassistant/components/notify/pushbullet.py:
```diff
@@ -150,14 +152,30 @@ def _push_data(self, message, title, data, pusher, tname=None):
+                    pusher.push_file(title=title, body=message,
+                                     file_name=file_url, file_url=file_url,
+                                     file_type=(mimetypes
+                                                .guess_type(file_url)[0]))
```
Would it be possible to replace line 164-173 with;
```
pusher.push_file(title=title, body=message, email=tname,
                           file_name=file_url, file_url=file_url,
                           file_type=(mimetypes
                                          .guess_type(file_url)[0]))
```
Since email=None should be the same as not using the argument?

tschmidty69 replied:
Sure, let me take a poke. I was just extending what was there but looking at it more, it looks like the whole thing can be simplified a great deal.

balloob committed:
Cloud Updates (#11404)

* Verify stored keys on startup

* Handle Google Assistant messages

* Fix tests

* Don't verify expiration when getting claims

* Remove email based check

* Lint

* Lint

* Lint

dfinlay committed:
Add support for the renaming of Yamaha Receiver Zones via configuration file. Added a test to cover the change, plus previously untested options. (#11402)

ryanm101 committed:
Plex api update (#11423)

* Updated plexapi to 3.0.5

* Removed removed server obj that was being used for the media URL generation as .url() is available in self._session
seasons() no longer exists -> renamed to season()
season() returns obj not list so corrected
season.index returns int, cast to str to zerofill

* Fixed Linting

daenny committed:
fix: hmip-etrv-2 now working with homeassistant (#11175)

* fix: hmip-etrv-2 now working with homeassistant (see also pull request at pyhomematic)

* fix linting issue and typo

* address comment @pvizeli

* Only use cached data in current operation mode

* fix linting issue

Commit:
Fix errors in zigbee push state (#11386)

* Fix errors in zigbee push state

* Fix formatting

* Added comment

andrey-git committed:
Add on/off supported feature to climate (#11379)

* Add on/off supported feature to climate

* Lint

cdce8p committed:
Fix CONF_FRIENDLY_NAME (#11438)

andrey-git committed:
Climate: fix missing "|" (#11441)

* Add on/off supported feature to climate

* Lint

* Fix missing |

* lint

balloob committed:
Google Assistant -> Google Actions (#11442)

rofrantz committed:
Added new climate component from Daikin (#10983)

* Added Daikin climate component

* Fixed tox & hound

* Place up the REQUIREMENTS var

* Update .coveragerc

* Removed unused customization

* Prevent setting invalid operation state

* Fixed hound

* Small refactor according to code review

* Fixed latest code review comments

* Used host instead of ip_address

* No real change

* No real change

* Fixed lint errors

* More pylint fixes

* Shush Hound

* Applied suggested changes for temperature & humidity settings

* Fixed hound

* Fixed upper case texts

* Fixed hound

* Fixed hound

* Fixed hound

* Removed humidity since even the device has the feature it cant be set from API

* Code review requested changes

* Fixed hound

* Fixed hound

* Trigger update after adding device

* Added Daikin sensors

* Fixed hound

* Fixed hound

* Fixed travis

* Fixed hound

* Fixed hound

* Fixed travis

* Fixed coverage decrease issue

* Do less API calls and fixed Travis failures

* Distributed code from platform to climate and sensor componenets

* Rename sensor state to device_attribute

* Fixed hound

* Updated requirements

* Simplified code

* Implemented requested changes

* Forgot one change

* Don't allow customizing temperature unit and take it from hass (FOR NOW)

* Additional code review changes applied

* Condensed import even more

* Simplify condition check

* Reordered imports

* Disabled autodiscovery FOR NOW :(

* Give more suggestive names to sensors

ciotlosm committed:
Updated generic thermostat to respect operation_mode and added away mode (#11445)

* Updated generic thermostat to respect operation_mode and added away mode

* Updated tests to include away mode and corrected init problem for sensor state
Added more tests to improve coverage and corrected again lint errors
Fixed new test by moving to correct package
Fixed bug not restoring away mode on restart

* Added support for idle on interface through state

* Added back initial_operation_mode and modified away_temp to be only one for now

* Fixed houndci-bot errors

* Added back check for None on restore temperature

* Fixed failing tests as well

* Removed unused definitions from tests

* Added use case for no initial temperature and no previously saved temperature

PhracturedBlue committed:
Reconnect to alarmdecoder on disconnect (#11383)

* Reconnect to alarmdecoder on disconnect

* Use track_point_in_time instead of call_later

* use alarmdecoder 1.13.2 which has a more robust reconnection fix

bakedraccoon committed:
Set tahoma cover scan interval to 60 seconds (#11447)

* Increase tahoma scan interval to 60

* import timedelta

* fix lint

balloob committed:
Add some tests to the cloud component (#11460)

cdce8p committed:
Input Select - Added service description (#11456)

cdce8p committed:
Input Boolean - Deleted 'DEFAULT_INITIAL' (#11453)

cdce8p committed:
Updated gitignore file (#11452)

titilambert committed:
Update hydroquebec component to use hass httpsession (#11412)

* Update hydroquebec component to use hass httpsession

* Remove blank line

Commit:
Catch everything when calling to OctoPrint API to fix #10557 (#11457)

Danielhiversen committed:
Update PULL_REQUEST_TEMPLATE.md (#11465)

* Update PULL_REQUEST_TEMPLATE.md

* Add period

mnoorenberghe committed:
iOS 10 should be served javascript_version:es5 (#11387)

* iOS 10 should be served javascript_version:es5

Fixes #11234

* Update min Safari version to 12

mnoorenberghe committed:
iOS 10 should be served javascript_version:es5 (#11387)

* iOS 10 should be served javascript_version:es5

Fixes #11234

* Update min Safari version to 12

balloob committed:
Version bump to 0.60.1

balloob committed:
Alexa to not use customize for entity config (#11461)

* Alexa to not use customize for entity config

* Test Alexa entity config

* Improve tests

* Fix test

balloob committed:
Merge pull request #11470 from home-assistant/release-0-60-1

0.60.1

balloob committed:
Merge branch 'master' into dev

Chris-V committed:
Don't duplicate html5 registrations (#11451)

* Don't duplicate html5 registrations

If a registration is posted and another registration with the same
endpoint URL exists, update that one instead. That way, we preserve
the device name that has been configured. The previous behavior used to
append 'unnamed device' registrations over and over, leading to
multiple copies of the same registration. The endpoint URL is unique per
service worker so it is safe to update matching registrations.

* Refactor html5 registration view to not write json in the event loop

hawk259 committed:
AlarmDecoder remove icon function as BinarySensorDevice handles it correctly now (#11467)

* remove icon function as BinarySensorDevice handles it correctly now

* removing _type, not used

fabaff committed:
Upgrade psutil to 5.4.3 (#11468)

fabaff committed:
Upgrade alpha_vantage to 1.8.0 (#11476)

balloob committed:
Fix vultr tests (#11477)

fabaff committed:
Upgrade yarl to 0.17.0 (#11478)

abondoe committed:
Add Touchline climate platform (#10547)

* Add toucline platform

* Fix bugs suggested by houndci-bot

* Fix bugs suggested by houndci-bot

* Fix style based on comments and lint

* Remove target temperature

* Fix unexpected EOF

* Fix unexpected EOF

* Fix wrongfully named numberOfDevices variable

* Add target temperature

* Update requirements_all.txt

* Change after review

* Add supported features
* Remove update in constructor
* Set member variables to none in constructor

* Minor changes

fabaff committed:
Upgrade pysnmp to 4.4.4 (#11485)

jesserockz committed:
Add new iGlo component (#11171)

* Add new iGlo component

* Missing comma
Add extra blank lines

* Dont change state in turn_on
Remove unused variables
Update before add

* Fixing some lint issues

fabaff committed:
Add missing configuration variables (#11390)

* Add missing configuration variables

* Minor changes

* Sync platforms and other minor updates

balloob committed:
Clean up Alexa.intent and DialogFlow.intent (#11492)

* Clean up Alexa.intent

* Restructure dialogflow too

* Lint

* Lint

tomwaters committed:
Fix time functions would throw errors in python scripts (#11414)

* Fix time functions would throw errors in python scripts

* Added unit test for time.strptime, change variable name to satisfy lint

* Added docstring for time attribute wrapper method to satisfy lint

* Fixed line too long lint problem

Chris-V committed:
Upgrade pywebpush to 1.5.0 (#11497)

This version includes a fix for the serialization errors that occurred
when updating push subscriptions.

Changes since version 1.3.0:
https://github.com/web-push-libs/pywebpush/compare/1.3.0...28d2b55f

JulianKahnert committed:
upgrade schiene to 0.20 (#11504)

Commit:
timer: include the remaining time in the state attributes (#11510)

Add the amount of remaining time before a timer is finished in its state attributes, so that it is received when fetching a timer state.

In my particular case, it allows me to correctly implement a UI for a timer when it is paused and when the UI didn't received the pause state change (which happens when the UI is started after the pause). In this case, it is impossible to show the remaining amount of time before the timer ends.

Julius2342 committed:
More tolerant KNX component if gateway cant be connected (#11511)

* Issue #11432: Do not stop initializing KNX when tunelling device cant be reached

*  Issue #11432: Mark devices as unavailable if gateway cant be connected

todschmidt committed:
Snips (new) added speech response, parse snips/duration (#11513)

* Snips changed to using <username>:intentName instead of user_IDSTRING__intentName

* added response to snips

* blank line

* added unittests

* houndy

* sdtuff

* more stuff

* Update test_snips.py

* Update snips.py

* Split log tests to avoid dict ordering in py34

* Update test_snips.py

* Update test_snips.py

* still broken

* fixed tests

* fixed tests

* removed fix submitted in another PR

DanNixon committed:
Add Dark Sky weather component (#11435)

amelchio committed:
Lazy loading of service descriptions (#11479)

* Lazy loading of service descriptions

* Fix tests

* Load YAML in executor

* Return a copy of available services to allow mutations

* Remove lint

* Add zha/services.yaml

* Only cache descriptions for known services

* Remove lint

* Remove description loading during service registration

* Remove description parameter from async_register

* Test async_get_all_descriptions

* Remove lint

* Fix typos from multi-edit

* Remove unused arguments

* Remove unused import os

* Remove unused import os, part 2

* Remove unneeded coroutine decorator

* Only use executor for loading files

* Cleanups suggested in review

* Increase test coverage

* Fix races in existing tests

balloob committed:
Fix canary flaky test (#11519)

DanNixon committed:
Address missed review comments for Dark Sky weather (#11520)

See #11435

cpcowart committed:
New features for Owntracks device_tracker (#11480)

* New features for Owntracks device_tracker

- Supporting a mapping of region names in OT to zones in HA, allowing
  separate namespaces in both applications. This is especially helpful
  if using one OT instance to update geofences for multiple homes.
- Creating a setting to ignore all location updates, allowing users to
  rely completely on enter and leave events. I have personally always
  used OT integrations with home automation this way and find it the
  most reliable.
- Allowing the OT topic to be overridden in configuration

* Fixing configuration of MQTT topic, related tests

* Tests for Owntracks events_only feature

* Tests for customizing mqtt topic, region mapping

* Fixing _parse and http for owntracks custom topic

* Making tests more thorough and cleaning up lint

aosadchyy committed:
Reconnect before mochad switch send command (#11296)

* Connection to mochad occasionally stalls on RPi and CM19A. Reconnect one switch send command.

* Formatting and exception hanling fixes

* Moved import inside the method. Logging outside the try-catch.

* Tailing whitespaces.

* MockDependency on pymochad in unit tests to resolve exceptions

* patch pymochad MochadException in unit tests to resolve exceptions

* patch pymochad MochadException in unit tests to resolve exceptions

* cleaned unused import

* lint issue fixed

* pylint issue fixed

florianj1 committed:
Fix Kodi channels media type (#11505)

* Update kodi.py

* Update kodi.py

Fix code style

ChristianKuehnel committed:
Extension sensor alpha_vantage (#11427)

* Extended sensor alpha_vantage

* Improved error handling if symbol is not found. Now we add the symbols that were found,
* Added option to give names and currencies to symbols.
* Added support to read foreign exchange information.
* Icons are selected based on the currency (where MDI has a matching icon)

* added missing line at the end of the file...

* renamed variable as required by pylint

* Fix typos, ordering, and style

Commit:
MQTT json attributes (#11439)

* MQTT json attributes

* Fix lint

* Amends following comments

* Fix lint

* Fix lint

* Add test

* Fix typo

* Amends following comments

* New tests

* Fix lint

* Fix tests

michaelkuty committed:
Support pushing all sensors and fix wrong metrics. (#11159)

For example all metrics with unit % match humidity.
This generate correct metrics like this
# HELP nut_ups_battery_charge sensor.nut_ups_battery_charge
# TYPE nut_ups_battery_charge gauge
nut_ups_battery_charge{entity="sensor.nut_ups_battery_charge",friendly_name="NUT UPS Battery Charge"} 98.0
nut_ups_battery_charge{entity="sensor.nut_ups_battery_charge_2",friendly_name="NUT UPS Battery Charge"} 97.0

yienxu committed:
Add options feature to Baidu TTS. (#11462)

* Add options feature to Baidu TTS.

Add options feature: supported_options() and default_options() added, get_tts_audio updated to accommodate options.

* Add options feature to Baidu TTS.

Add options feature: supported_options() and default_options() added, get_tts_audio updated to accommodate options.

* Fix style

* Fix style

Change the order of content of lists and dictionaries.

* Fix style

Changed order of imports, and fixed grammar errors.

camrun91 committed:
Insteon local update (#11088)

* trying to rework device discovery. now the main component will do the getlinked and pass it to the sub-components. no longer any config needed other than what is needed to connect to the hub. device names are no longer stored. core team told us to stop using configurator to ask for names. there should be a way to set names in hass...possibly this https://home-assistant.io/docs/configuration/customizing-devices/

* fix device types

* make device names just be the isnteon device id

* revert some config changes

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* update insteon client

* linting fixes

* Error Clean up

* Update to make requested changes

* more changes

* Finish requested changes to components

* Fixing Rebase Conflicts

* fix device types

* make device names just be the isnteon device id

* revert some config changes

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* update insteon client

* linting fixes

* Error Clean up

* Update to make requested changes

* more changes

* Finish requested changes to components

* Update Insteon_Local for performance improvements

* Fix errors from get_linked

* Fix typo

* Requested changes

* Fix spacing

* Clean up

* Requested Changes

todschmidt committed:
Snips: (fix) support new intentName format (#11509)

* support new intentName format

* Added tests for new and old format names

* pylint warning fixes

fabaff committed:
Upgrade numpy to 1.14.0 (#11542)

fabaff committed:
Upgrade python-etherscan-api to 0.0.2 (#11535)

fabaff committed:
Upgrade Sphinx to 1.6.6 (#11534)

swilson committed:
ZoneMinder event sensor updates (#11369)

* Switch the ZoneMinder event sensor over to use the consoleEvents API, and add monitored_conditions for 'hour', 'day', 'week', 'month' and 'all'. 'all' is enabled by default to provide backward compatibility with the old behaviour.

* - Switch to new string formatting
- Remove redundant validator

* De-lint

balloob committed:
Move Google Assistant entity config out of customize (#11499)

* Move Google Assistant entity config out of customize

* CONF_ALIAS -> CONF_ALIASES

* Lint

fabaff committed:
Upgrade lightify to 1.0.6.1 (#11545)

cdce8p committed:
Improved test runtime (#11553)

* Remove redundent assert statements and cleanup

* Added 'get_date' function

* Replace 'freeze_time' with 'mock.patch'
* Tox in 185s (py35)

* Removed obsolete 'freeze_time' from test_updater
* Tox 162s (py35)

* Remove test requirement 'freezegun'

* Fixed flake8 errors

* Added 'mock.patch' for 'feedparser.parse'

* Made 'FUNCTION_PATH' a constant

* Remove debug statements.

ericpignet committed:
Add 2 media_player services and 1 custom service to Squeezebox platform (#10969)

* Add 2 media_player services and 1 custom service to Squeezebox platform

* Fix pylint error

* Remove apostrophe in example

* Split method into command and parameters

* Fix Lint error

philk committed:
Fix error on entity_config missing (#11561)

If the `google_assistant` key exists in the config but has no
`entity_config` key under it you'll get an error.

```
  File "/Users/pkates/src/home-assistant/homeassistant/components/google_assistant/http.py", line 51, in is_exposed
    entity_config.get(entity.entity_id, {}).get(CONF_EXPOSE)
AttributeError: 'NoneType' object has no attribute 'get'
```

fabaff committed:
Upgrade keyring to 10.3.2 (#11531)

nkgilley committed:
Coinbase.com sensor platform (#11036)

* coinbase sensors

use hass.data, load_platform

* add exchange rate sensors

dont pass complex object over event bus

* check for auth error

andreacampi committed:
Try to fix crashes after Hue refactoring (#11270)

* Try to fix crashes after Hue refactoring

Refs #11183

* Fix usage of dispatcher_send via helper.

* Address review feedback.

balloob committed:
Remove execution file perm (#11563)

amelchio committed:
Deprecate explicit entity_id in template platforms (#11123)

* Deprecate explicit entity_id in template platforms

* Use config validator for deprecation

* Fix copy/paste typos

* Also print the config value

* Add test for config validator

* Assert the module name that logged the message

balloob committed:
Update frontend to 20180110.0

balloob committed:
Update frontend to 20180110.0

balloob committed:
Test tweaks (#11560)

* Fix is_allowed_path on OS X

* Move APNS2 inside func in test

bakedraccoon committed:
Add Velux Windows to Tahoma (#11538)

* Add Velux Windows to Tahoma

* fix linit

* add supported Tahoma devices by @glpatcern

* hound

* lint

* fix logging

* lint

* lint

* remove blank line after docstring

* changes based on notes by @armills

* fix logging

pvizeli committed:
Extend hass.io services / updater (#11549)

* Extend hass.io services

* Add warning for carfuly options with hass.io

* update tests

* finish tests

* remove update calls

* address comments

* address comments p2

* fix tests

* fix tests

* Use token also for proxy

* Add test for server_host

* Fix test

* Fix tests

* Add test for version

* Address comments

andrey-git committed:
Add sensibo_assume_state service to Sensibo climate (#11567)

randellhodges committed:
Tracking all groups to allow changing of existing groups (#11444)

* Tracking all groups to allow changing of existing groups

* Unit tests

* Fix flake8 warnings in test

amelchio committed:
Fix new squeezebox service descriptions for lazy loading (#11574)

DanNixon committed:
Mark REST binary_sensor unavailable if request fails (#11506)

* Mark REST binary_sensor unavailable if request fails

* Add test suite for RESTful binary sensor

andersonshatch committed:
Tado improvements - hot water zone sensors and climate precision (#11521)

* Add tado hot water zone sensors

* Set precision to match tado app/website

ulido committed:
Make the rpi_rf component thread-safe using an RLock (#11487)

* Make the rpi_rf component thread-safe

The previous implementation suffered from race conditions when two rpi_rf switches are triggered at the same time. This implementation uses an RLock to give one thread at a time exclusive access to the rfdevice object.

* cleanup

* fix lint

danielperna84 committed:
Update pyhomematic, support new devices (#11578)

hawk259 committed:
Alarmdecoder add validation of the zone types (#11488)

* Alarmdecoder add validation of the zone types

* fix line length

emlove committed:
Support OSRAM lights on ZHA (#11522)

swilson committed:
Fix bluetooth tracker source (#11469)

* Add new sources for bluetooth and bluetooth_le trackers (rather than just using the generic 'gps' catch-all).

* Fix lint issues

kennedyshead committed:
Fix for asuswrt, telnet not working and presence-detection for router mode (#11422)

* Fix for telnet mode
Fix for presence detection

closes #11349 #10231

* Optimisation when device is disconnected

balloob commented:
Renaming your PR to add [WIP] until you have simplified it. Remove the [WIP] prefix if you want us to look at it again

andrey-git committed:
Remove aux_heat support from Sensibo now that UI supports on/off (#11579)

Laqoore committed:
Changed device type of media player and cover to switch (#11483)

* Changed device type of media player and cover to switch

Covers and media players should not be of device type 'light'. Example: If user requests all lights to switch to off, covers are closed and media players are affected too.

* Fix test

rwa committed:
Control ordering of display in history component (#11340)

* Make the order shown in the history component match the ordering given in the configuration of included entities (if any)

* return the sorted result

* optimize sorting.  since entities only appear once, we can break from a search on first find, and no copy of the list is needed

balloob committed:
Pr/11430 (#11587)

* Fix error when name is non-latin script

When the name is non-latin script (e.g. Japanese), slugify returns empty string and causes failure in a later stage. This commit fixes the issue by using default name.

* Add test

cgtobi committed:
Cast return values and add unit tests for the yahoo weather component. (#10699)

* Add unittest for yahoo weather component.

* Add requitements for tests.

* Update requirements.

* Revert "Update requirements."

This reverts commit 024b5cf8bf0d1bdfe927a117f5befb53c5a5cff4.

* Update requirements.

* Randomize sample data.

* Remove unnecessary methods.

* Remove dependency and replace with MockDependency.

* Add temporary fix for yahoo weather API issue.

bakedraccoon committed:
add velux roller shutter to tahoma (#11586)

With #11538 I forgot to add another type of Roller shutters that should be supported.

abjorshammar committed:
Added support for enable/disable motion detection (#11583)

* Added support for enable/disable motion detection

* Changed name of variable for exception, lint error

* Moved motion_detection_enabled property to the other properties

todschmidt committed:
Simplified push calls

todschmidt committed:
Simplified push calls

todschmidt reviewed (commented):
Cleaned up. Email=None did not work with the pushbullet API but got around it using dummy keyword args; looks much cleaner now.

tschmidty69 commented:
Ugh, munged up the rebase and too tired to fix it, recreated it
